### ISSUE TITLE
Adding ability to exclude some parsers

### ISSUE THREAD

**Issue by JinneeJ** [atlassian/commonmark-java#43]
Hi! Is it possible to add an ability (via Builder) to disable some parsers? There could be cases when people want to parse everything except lists for example. Or for some cases people need to parse only bold and italic. It would be useful to have only one lib for all these cases :)

**Comment by robinst** [comment]
Yeah, I think that would be nice!

In terms of implementation, I think it would be good to not expose the specific block parsers (e.g. `ListBlockParser`). Instead, the API could take node types (e.g. `ListBlock.class`) and we would do the correct mapping internally.

So, to clients, the API would look something like this:

```java
Parser parser = Parser.builder()
        .exclude(ListBlock.class)
        .build();
```

What do you think? Do you want to create a PR for that?

**Comment by kirys** [comment]
What about something like an allowoed nodes list instead?

**Comment by lalunamel** [comment]
We've made a PR for this issue. https://github.com/atlassian/commonmark-java/pull/67

We added a method, `allowedBlockTypes` to the Parser builder that takes a list of `Block` classes to use for parsing.

```
/**
* Describe the list of markdown features the parser will recognize and parse.
*
* By default, Commonmark will recognize and parse the following set of core markdown features:
*
* Heading (#) - Heading.class
* HTML (<html></html>) - HtmlBlock.class
* Horizontal Rule / Thematic Break (---) - ThematicBreak.class
* Fenced Code Block (```) - FencedCodeBlock.class
* Indented Code Block - IndentedCodeBlock.class
* Block Quote (>) - BlockQuote.class
* Ordered / Unordered List (>) - ListBlock.class
*
* To parse only a subset of the features listed above, pass a list of each feature's associated Node class.
* E.g., to only parse Headings and Lists:
* <pre>
*     {@code
*     Parser.builder().allowedBlockTypes(Heading.class, ListBlock.class);
*     }
* </pre>
*
* @param allowedBlockTypes A list of nodes the parser will parse.
*                     If this list is empty, the parser will not recognize any Commonmark core markdown features.
*
* @return {@code this}
*/
```

**Comment by kirys** [comment]
well but it seems that the status of this is stuck :(

**Comment by robinst** [comment]
@kirys I've added some feedback to the pull request, but unfortunately have not heard anything from @lalunamel yet.

**Comment by evpaassen** [comment]
PR #67 seems to cover only block types. But what if one would like to disable, for example images (because these are external resources)? These are inline.

**Comment by evpaassen** [comment]
> But what if one would like to disable, for example images (because these are external resources)? These are inline.

Oh, apparently, there's #69 now.

**Comment by robinst** [comment]
Released in 0.9.0 with some other changes: https://github.com/atlassian/commonmark-java/releases/tag/commonmark-parent-0.9.0
